courses/programming-in-haskell/pih-exercises/src/Chapter06.hs

```diff
@@ -0,0 +1,41 @@
+-- | Exercises for chapter 5.
+
+module Chapter06 (and', concat', replicate', (!!!), elem', merge, msort) where
+
+-- * Exercise 1
+and' :: [Bool] -> Bool
+and' [] = True
+and' (False:xs) = False
+and' (True:xs) = and' xs
+
+concat' :: [[a]] -> [a]
+concat' [] = []
+concat' (xs:xxs) = xs ++ concat' xxs
+
+replicate' :: Int -> a -> [a]
+replicate' n x | n == 0 = []
+               | n > 0 = x:(replicate' (n-1) x)
+
+(!!!) :: [a] -> Int -> a
+(!!!) (x:xs) n
+    | n == 0 = x
+    | n > 0 = (!!!) xs (n-1)
```

Comment:
Remember that you can use infix notation if you think that helps readability, i.e. `xs !!! (n - 1)`.